lightning/src/routing/gossip.rs

```diff
@@ -570,6 +570,12 @@ where C::Target: chain::Access, L::Target: Logger
 			action: ErrorAction::IgnoreError,
 		})
 	}
+
+	fn provided_init_features(&self, _their_node_id: &PublicKey) -> InitFeatures {
+		let mut features = InitFeatures::empty();
+		features.set_gossip_queries_optional();
```

Comment:
Should this also set `initial_routing_sync` since that is cleared in `known_channel_features`? Maybe it would be better to define `known_channel_features` as the set difference of `InitFeatures::known` with `P2PGossipSync::provided_init_features` (as a const somewhere). Then what is described in the comment on `InitContext`'s optional features wouldn't be necessary.

Reply:
Should we call this out in the commit message and changelog since it is a behavioral change? Will we ever set `initial_routing_sync` now?

Reply:
Any way we could still do something with set difference, though? Would need to special case removing `initial_routing_sync`, but I guess that is an unordinary feature anyhow.

Reply:
Discussed offline. My primary concern was to avoid relying on a comment to know when to update `known_channel_features`. Possible alternative could be to have a unit test asserting that the intersection of `ChannelManager` and `P2PGossipSync`'s provided features is empty.